ts/Extensions/Annotations/Controllables/ControllableEllipse.ts

```diff
@@ -0,0 +1,304 @@
+        cy: 'cy'
+    });
+
+    /* *
+     *
+     *  Constructors
+     *
+     * */
+
+    public constructor(annotation: Annotation, options: EllipseShapeOptions, index: number) {
+        this.angle = options.angle;
+        this.init(annotation, options, index);
+        this.collection = 'shapes';
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public addControlPoints = ControllableMixin.addControlPoints;
+    public anchor = ControllableMixin.anchor;
+    public attr = ControllableMixin.attr;
+    public attrsFromOptions = ControllableMixin.attrsFromOptions;
+    public destroy = ControllableMixin.destroy;
+    public getPointsOptions = ControllableMixin.getPointsOptions;
+    public linkPoints = ControllableMixin.linkPoints;
+    public point = ControllableMixin.point;
+    public scale = ControllableMixin.scale;
+    public setControlPointsVisibility = ControllableMixin.setControlPointsVisibility;
+    public shouldBeDrawn = ControllableMixin.shouldBeDrawn;
+    public transform = ControllableMixin.transform;
+    public translatePoint = ControllableMixin.translatePoint;
+    public update = ControllableMixin.update;
+    public angle: number = void 0 as any;
+    public referencePoints: Array<ReferencePointsOptions> = void 0 as any;
+
+    /**
+     * @type 'ellipse'
+     */
+    public type = 'ellipse';
+
+    public translate = ControllableMixin.translateShape;
+
+    /**
+     * Functions
+     */
+
+    /**
+     * Transform the middle point (center of an ellipse).
+     * Mostly used to handle dragging of the ellipse.
+     */
+    public transformPoint(): void {
+        // Call save points, to handle the pinning of the angle
+        // and radius to axes points.
+        this.savePoints();
+        ControllableMixin.transformPoint.apply(this, arguments);
+    }
+
+    public init(annotation: Annotation, options: EllipseShapeOptions, index: number): void {
+        ControllableMixin.init.call(this, annotation, options, index);
+        this.savePoints();
+    }
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public render(parent: SVGElement): void {
+        const attrs = this.attrsFromOptions(this.options);
+
+        const graphic = this.annotation.chart.renderer.createElement('ellipse');
+        graphic.attr(attrs).add(parent);
+        this.graphic = graphic;
+        ControllableMixin.render.call(this);
+    }
+
+    public redraw(animation?: boolean): void {
+        const position = this.anchor(this.points[0]).absolutePosition,
+            attrs = this.getAttrsFromPoints();
+        if (position) {
+            this.graphic[animation ? 'animate' : 'attr']({
+                cx: attrs.cx,
+                cy: attrs.cy,
+                rx: attrs.rx,
+                ry: attrs.ry,
+                transform: `rotate(${attrs.angle}, ${position.x}, ${position.y})`
+            });
+        } else {
+            this.graphic.attr({
+                x: 0,
+                y: -9e9
```

Comment:
Why such a high number here?

Reply:
It is the same as in the ControllableCircle.ts, so I decided to keep to the pattern
https://github.com/highcharts/highcharts/blob/master/ts/Extensions/Annotations/Controllables/ControllableCircle.ts#L126